frontend/src/pages/notebookController/StartServerModal.tsx

```diff
@@ -15,93 +29,183 @@ type StartServerModalProps = {
 type SpawnStatus = {
   status: AlertVariant;
   title: string;
-  reason: React.ReactNode;
+  description: React.ReactNode;
 };
 
-const StartServerModal: React.FC<StartServerModalProps> = ({
-  notebook,
-  startShown,
-  setStartModalShown,
-  onClose,
-}) => {
-  const [spawnStatus, setSpawnStatus] = React.useState<SpawnStatus | null>(null);
-
-  const isNotebookRunning = checkNotebookRunning(notebook);
-  const notebookLink = notebook?.metadata.annotations?.['opendatahub.io/link'];
-
-  React.useEffect(() => {
-    let timer;
-    if (isNotebookRunning) {
-      setSpawnStatus({
-        status: AlertVariant.success,
-        title: 'Success',
-        reason: 'The notebook server is up and running. This page will update momentarily.',
-      });
-      timer = setTimeout(() => {
-        if (notebookLink) {
-          window.location.href = notebookLink;
-        } else {
+const StartServerModal: React.FC<StartServerModalProps> = React.memo(
```

Comment:
Help me understand why you made this `React.memo`?

Reply:
I thought we could optimize the performance to avoid useless rendering by using `React.memo`, but after I check the code, it looks like we don't need it here.

Reply:
Proactively trying to get ahead of performance problems is not ideal (usually ever). Unless you know something is going to be inherently non performant, you should let React do its thing. React claims to have a handle on the day-to-day performance of all component renderings.

If you see spamming of network calls, the screen flickering between multiple states, the screen freezing for moments, or something along those lines -- you should avoid optimizing.

`useMemo` should also not be done willy nilly... if you need to enforce a slower recompute of an object reference that's a good use of `useMemo` -- otherwise you should avoid it too unless it is needed for some specific reason.